node/node.go

```diff
@@ -0,0 +1,80 @@
+package node
+
+import (
+	"context"
+
+	logging "github.com/ipfs/go-log/v2"
```

Comment:
> and then stick it on the node?

As discussed in https://github.com/celestiaorg/celestia-node/issues/39, the second approach is more appealing.

Reply:
> do we have to import IPFS logging? Can we just use something like this:

I don't like the fact that we would need to import whole beefy `go-ethereum` for that. On the other hand, `go-log` is designed to be used not just in `go-ipfs` and extracted into a separate module. Also, we would use `go-log/v2` anyway as we depend on libp2p and ipfs repos, so that's inevitable. However, that's not an issue IMO, `go-log` is just a tiny convenient wrapper over https://github.com/uber-go/zap

Reply:
No we don't have to import geth but we can do our own thing. Why not logrus or something like that?

Reply:
https://github.com/sirupsen/logrus

Reply:
@renaynay, the difference between `logrus` and `go-log(zap)` is that the former treats the whole log waterfall as it comes from one app, while the latter splits all log entries into modules(loggers). 
In the case of this PR, we have only a node logger for now:
![image](https://user-images.githubusercontent.com/22449364/131837529-32df361a-7aa6-4253-a160-f5bb79521e5d.png)

In the case with logrus, we would be able to do so by using key-value entries and this won't look good:
![image](https://user-images.githubusercontent.com/22449364/131837641-85c96ff8-7c6f-460a-aa5c-5c8b409d92a4.png)

Another thing, I am not really sure how to hijack logs coming from libp2p to another logger and I feel like this is not possible at all, so by using another logger (not zap) we would end having two at the same time, what is not desirable.

Also, what's the problem with zap? Have you tried it or is that an allergy to IPFS things 😄?